ARMmbed/meta-mbl
# Clean up LICENSE variables in atf recipes

**Initial commit message**
Clean up LICENSE variables in atf recipes

* 'LICENSE-uboot.cfgout = "GPLv2"' doesn't mean anything useful to
  BitBake: BitBake doesn't deal with licenses at an individual file
  level. Additionally, on raspberrypi3 this file isn't added so the
  GPLv2 license doesn't apply there. Remove this line and remove "GPLv2"
  from the value of the LICENSE variable set in atf.inc. In
  atf-imx7s-warp.bb add a line to append GPLv2 to LICENSE when
  uboot.cfgout is added to SRC_URI.

* 'LICENSE-atf = "BSD-3-Clause"' doesn't mean anything useful to BitBake
  either so remove it.

For IOTMBL-1025: [meta-mbl#316] atf-warp7 recipe seems to break Warp7
Incremental Builds

**Jenkins build**
* http://jenkins.mbed-linux.arm.com/view/jh/job/jh-test/8/ (Failed due to unrelated issue `No recipes available for: /work/machine-raspberrypi3-mbl/mbl-manifest/build-mbl/conf/../../layers/meta-virtualization/recipes-kernel/linux/linux-yocto_4.14.bbappend`).
* http://jenkins.mbed-linux.arm.com/view/jh/job/jh-test/9/ (second attempt; succeeded).
* http://jenkins.mbed-linux.arm.com/view/jh/job/jh-test/14/ (after changes for review; now with pico build; w7 & rpi3 succeeded; pico failed with same error as vanilla master Jenkins build).

**Testing done**
* Build and run locally for imx7s-warp-mbl
* Check the license manifest (build-mbl/tmp-mbl-glibc/deploy/licenses/mbl-console-image-imx7s-warp-mbl-20181221163225/image_license.manifest) reports atf's license correctly for an imx7s-warp-mbl build:
  ```
  RECIPE NAME: atf-imx7s-warp-mbl
  VERSION: git
  LICENSE: BSD-3-Clause & Apache-2.0 & GPL-2.0
  FILES: rot_key.pem atf-bl2-fip.bin
  ```
* Build and run locally for raspberrypi3-mbl
* Check the license manifest (build-mbl/tmp-mbl-glibc/deploy/licenses/mbl-console-image-raspberrypi3-mbl-20181221171327/image_license.manifest) reports atf's license correctly for a raspberrypi3-mbl build:
  ```
  RECIPE NAME: atf-raspberrypi3-mbl
  VERSION: git
  LICENSE: BSD-3-Clause & Apache-2.0
  FILES: rot_key.pem armstub8.bin
  ```



**Request for reviewers:** I've removed GPLv2 from atf's LICENSE on raspberrypi3 because it has no uboot.cfgout in SRC_URI. Is this definitely correct?
